cli/src/action/database/mod.rs

```diff
@@ -63,6 +77,227 @@ impl Action for MigrateAction {
     }
 }
 
+impl From<Box<dyn Error>> for CliError {
+    fn from(e: Box<dyn Error>) -> CliError {
+        CliError::ActionError(format!("encountered unexpected error: {}", e))
+    }
+}
+
+pub struct MigrateFromYamlAction;
+
+impl MigrateFromYamlAction {
+    /// Get the splinter state directory from the environment
+    fn env_state_dir() -> Result<String, CliError> {
+        let osstr = SplinterEnvironment::load()
+            .get_state_path()
+            .into_os_string();
+        match osstr.to_str() {
+            Some(s) => Ok(s.to_string()),
+            None => Err(CliError::ActionError(format!(
+                "could not format splinter environment path into string: {}",
+                osstr.to_string_lossy()
+            ))),
+        }
+    }
+
+    /// Get an admin service store to the specified database
+    fn db_admin_service_store(url: String) -> Result<Box<dyn AdminServiceStore>, CliError> {
+        let connection_uri = url.parse().map_err(|_| {
+            CliError::ActionError(format!("invalid database url provided: {}", url))
+        })?;
+
+        let store_factory =
+            splinter::store::create_store_factory(connection_uri).map_err(|err| {
+                CliError::ActionError(format!("failed to initialize store factory: {}", err))
+            })?;
+        Ok(store_factory.get_admin_service_store())
+    }
+
+    fn import_command(
+        stdout: &mut dyn Write,
+        state_dir: &str,
+        get_db_store: impl FnOnce() -> Result<Box<dyn AdminServiceStore>, CliError>,
+    ) -> Result<(), CliError> {
+        writeln!(stdout, "Upgrading splinterd state")?;
+
+        write!(stdout, "Loading YAML datastore... ")?;
+        fn invalid_utf8() -> CliError {
+            CliError::ActionError("'state_dir' is not a valid UTF-8 string".to_string())
+        }
+
+        let circuits_location = Path::new(&state_dir).join("circuits.yaml");
```

Comment:
The state files can be added as const above.

Reply:
Done!